examples/statistics/errorbars_and_boxes.py

```diff
@@ -0,0 +1,53 @@
+'''
+Example to create boxes from error using PatchCollection
+'''
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.collections import PatchCollection
+from matplotlib.patches import Rectangle
+
+# Number of data points
+n = 5
+
+# Dummy data
+x = np.arange(0, n, 1)
+y = np.random.rand(n)*5.
+
+# Dummy errors (above and below)
+xerr = np.random.rand(2, n)
+yerr = np.random.rand(2, n)
+
+# Create figure and axes
+fig, ax = plt.subplots(1)
+
+# Plot data points
+ax.errorbar(x, y, xerr=xerr, yerr=yerr, fmt='None', ecolor='k')
+
+
+def makeErrorBoxes(xdata, ydata, xerror, yerror, fc='r', ec='None', alpha=0.5):
+    '''
+    Function to create error boxes
+    '''
+
+    # Create list for all the error patches
+    errorboxes = []
+
+    # Loop over data points; create box from errors at each point
+    for xc, yc, xe, ye in zip(xdata, ydata, xerror.T, yerror.T):
+        rect = Rectangle((xc-xe[0], yc-ye[0]), xe.sum(), ye.sum())
+        errorboxes.append(rect)
+
+    # Create patch collection with specified colour/alpha
+    pc = PatchCollection(errorboxes, facecolor=fc, alpha=alpha, edgecolor=ec)
+
+    # Add collection to axes
+    ax.add_collection(pc)
+
+# Call function to create error boxes
+makeErrorBoxes(x, y, xerr, yerr)
```

Comment:
I think examples should always be functions so that users can copy-paste
use them.

On Thu, Jun 16, 2016, 15:22 Benjamin Root <notifications@github.com> wrote:

> In examples/statistics/errorbars_and_boxes.py
> <https://github.com/matplotlib/matplotlib/pull/6596#discussion_r67408151>:
>
> > +    # Create list for all the error patches
> > +    errorboxes = []
> > +
> > +    # Loop over data points; create box from errors at each point
> > +    for xc, yc, xe, ye in zip(xdata, ydata, xerror.T, yerror.T):
> > +        rect = Rectangle((xc-xe[0], yc-ye[0]), xe.sum(), ye.sum())
> > +        errorboxes.append(rect)
> > +
> > +    # Create patch collection with specified colour/alpha
> > +    pc = PatchCollection(errorboxes, facecolor=fc, alpha=alpha, edgecolor=ec)
> > +
> > +    # Add collection to axes
> > +    ax.add_collection(pc)
> > +
> > +# Call function to create error boxes
> > +makeErrorBoxes(x, y, xerr, yerr)
>
> should we be making this boxes *before* calling errorbar() so that the
> boxes appear below the errorbars?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/6596/files/7f412e8d59060ce61af0f111d70bc42c5c15985e#r67408151>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMMhXKKiv5idEDU6eJ_NxIHyM93Jpj1ks5qMaKJgaJpZM4I3grL>
> .
>